src/v3/plugins/github/urlIdParse.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+export function reviewUrlToId(url: string): string {
+  const result = new RegExp(
+    "^https://github.com/\\w+/\\w+/pull/\\d+#pullrequestreview-(\\d+)$"
+  ).exec(url);
+  if (result == null) {
+    throw new Error(`Error parsing review url ${url}`);
```

Comment:
I’m surprised that exceptions are the right semantics. Have you
considered returning `null` in the case where there is no match? Having
a URL that does not refer to a comment is not really an exceptional
condition.

Reply:
My intention is to use this function when generating GitHub addresses from the GitHub graphql response. In this case, you have a JSON that you know definitely corresponds to (say) a review comment, so you call `reviewCommentUrlToId` on that json's url. This api is appropriate for that case - if anything goes wrong in getting the id out of the review comment's url, it's cause for an error. So the contract  for the function is that it should take in review comment urls and nothing else.

If we start to need the function for other things we can upgrade its scope, but if we start using it e.g. to sniff arbitrary urls and figure out what they are, at that point we need a fairly different return signature anyway.

Reply:
Understood. I had the `findReferences` context in mind, but you’re right
that the signature (and implementation) would need to be different
anyway.